packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,151 @@
+    const intervalMinutes = props.interval.toMinutes().toString();
+    const percentage = props.percentage.toString();
+
+    let routingConfig;
+    let customDeploymentConfigName;
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: intervalMinutes,
+          canaryPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaCanary' + percentage+ 'Percent' + intervalMinutes + 'Minutes';
+    } else if (props.type == CustomLambdaDeploymentConfigType.LINEAR) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedLinear: {
+          linearInterval: intervalMinutes,
+          linearPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaLinear' + percentage + 'Percent' + intervalMinutes + 'Minutes';
+    }
+
+    // What the config is called in the AWS console
+    // We tack the id on so that, if the user has multiple configs with identical settings,
+    // deleting one will not affect the other
+    const resourceName = id + customDeploymentConfigName;
+    this.deploymentConfigName = resourceName;
+    this.deploymentConfigArn = arnForDeploymentConfig(resourceName);
+
+    // The AWS Custom Resource that calls CodeDeploy to create and delete the resource
+    new AwsCustomResource(this, 'DeploymentConfig', {
+      onCreate: { // Run on creation only, to make the resource
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        // This `resourceName` is the initial physical ID of the config
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
+      onUpdate: { // Run on stack update
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        // If `resourceName` is different from the last stack update (or creation),
+        // the old config gets deleted and the new one is created
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
+      onDelete: { // Run on deletion, or on resource replacement
+        service: 'CodeDeploy',
+        action: 'deleteDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+        },
+      },
+      // Least permissions, only have permission to create or delete this exact config
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: [this.deploymentConfigArn],
+      }),
+    });
+  }
+
+  private validateParameters(props: CustomLambdaDeploymentConfigProps): void {
+    if (!Number.isInteger(props.percentage)) {
```

Comment:
Any sense validating that this percentage is between some min and max value...?

Reply:
Yes. Per the CodeDeploy console, "Step percentage must be an integer between 1 and 99".